src/deserializer.rs

```diff
@@ -110,6 +110,26 @@ impl Deserializer {
         });
     }
 
+    pub fn read_bit_string(&mut self) -> DeserializationResult<BitString> {
+        return self._read_with_tag(3, |data| {
+            if data.is_empty() {
```

Comment:
An alternate way of writing this without using indexing (which is basically `unwrap` in my view):

```rust
let padding_bits = match data.get(0) =>
    None => return Err(DeserializationError::InvalidValue),
    Some(bits) => bits,
};
```


https://doc.rust-lang.org/nightly/std/primitive.slice.html#method.get